### ISSUE TITLE
Error in readLines(process_get_error_connection(self, private), ...) 

### ISSUE THREAD

**Issue by stla** [rstudio/shinytest#113]
Hello, 

 I've just installed `shinytest`. I get this error:

```
> library(shinytest)
> recordTest(".")
Error in readLines(process_get_error_connection(self, private), ...) : 
  'con' is not a connection
```

```
> sessionInfo()
R version 3.3.3 (2017-03-06)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=French_Belgium.1252  LC_CTYPE=French_Belgium.1252   
[3] LC_MONETARY=French_Belgium.1252 LC_NUMERIC=C                   
[5] LC_TIME=French_Belgium.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shinytest_1.1.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.11         magrittr_1.5         debugme_1.0.2       
 [4] xtable_1.8-2         showimage_1.0.0      R6_2.2.2            
 [7] rematch_1.0.1        webdriver_1.0.2.9000 httr_1.2.1          
[10] tools_3.3.3          parsedate_1.1.3      png_0.1-7           
[13] withr_1.0.2          htmltools_0.3.6      assertthat_0.2.0    
[16] digest_0.6.12        crayon_1.3.2         processx_2.0.1      
[19] shiny_1.0.3.9001     base64enc_0.1-3      testthat_1.0.2      
[22] curl_2.7             mime_0.5             pingr_1.1.2         
[25] jsonlite_1.5         httpuv_1.3.5 
```

**Comment by bstabler** [comment]
It appears I can't create a new process.  Maybe the issue is the same as https://github.com/r-lib/processx/issues/61?

```
> library(processx)
> run("echo","hi")
$status
[1] 0

$stdout
[1] "hi\n"

$stderr
[1] ""

$timeout
[1] FALSE

> p = process$new("echo","hi")
> p$is_alive()
[1] FALSE
> p$read_output_lines()
Error in readLines(process_get_output_connection(self, private), ...) : 
  'con' is not a connection
> p$read_error_lines()
Error in readLines(process_get_error_connection(self, private), ...) : 
  'con' is not a connection
```

**Comment by wch** [comment]
I suspect that processx is not starting the process correctly in Windows for some reason.

I can start phantomjs from the command line with:

```
"C:\\Users\\IEUser\\AppData\\Roaming/PhantomJS/phantomjs.exe" --proxy-type=none --webdriver=127.0.0.1:8290 --webdriver-loglevel=INFO
```

(Note that this prints out a message `phantomjs://platform/console++.js:263 in error`, but this doesn't seem to cause any problems; it also does this on my Mac.)

However, if I start up the process this way, it dies:

```R
library(processx)
cmd <- '"C:\\Users\\IEUser\\AppData\\Roaming/PhantomJS/phantomjs.exe" --proxy-type=none --webdriver=127.0.0.1:8292 --webdriver-loglevel=INFO'
p <- process$new(commandline = cmd)
p$is_alive() # TRUE
Sys.sleep(1)
p$is_alive() # FALSE
```

**Comment by wch** [comment]
Removing the `"` characters around the path to `phantomjs.exe` solves the startup issue.

```
library(processx)
cmd <- 'C:\\Users\\IEUser\\AppData\\Roaming/PhantomJS/phantomjs.exe --proxy-type=none --webdriver=127.0.0.1:8292 --webdriver-loglevel=INFO'
p <- process$new(commandline = cmd)
p$is_alive() # TRUE
Sys.sleep(1)
p$is_alive() # TRUE
```

Of course, this won't work if there's a space in the path, so the solution will be to change `run_phantomjs` to start it using `command` and `args` instead of `commandline`.

**Comment by wch** [comment]
@stla, @bstabler: I just pushed a fix in the webdriver package. Can you try it with:

```
# Restart R
install_github("rstudio/webdriver")
```

**Comment by stla** [comment]
Hi, I get this error now:
```
> library(shinytest)
> recordTest()
Error in sd_startShiny(self, private, path) : 
  Cannot find shiny port number. Error: '\"C:/PortableApps/R-portable_3.3.3/App/R-Portable/bin/x64/R.exe\"' is notrecognized as an internal or external command,operable program or batch file.
```

**Comment by wch** [comment]
@stla, that's due a similar issue in shinytest. I've just pushed a fix. Can you try it out?

```R
# Restart R
install_github("rstudio/shinytest")
```

**Comment by bstabler** [comment]
I get the same error

```
> library(shinytest)
> #library(webdriver)
> #webdriver::run_phantomjs()
> recordTest("C:/projects/development/VisionEval/sources/VEGUI/app.R")
Error in sd_startShiny(self, private, path) : 
  Cannot find shiny port number. Error: '\"C:/PROGRA~1/R/R-33~1.2/bin/x64/R.exe\"' is not recognized as aninternal or external command,operable program or batch file.
```

but this now appears to work

```
> webdriver::run_phantomjs()
$process
PROCESS 'PHANTO~1.EXE', running, pid 5540.

$port
[1] 7148
```

**Comment by wch** [comment]
@bstabler, see my previous message about reinstalling shinytest.

**Comment by bstabler** [comment]
It's working now, thanks!

**Comment by stla** [comment]
Yes it's working. Thanks.

**Comment by wch** [comment]
Great!

Note: fixed in 3f11662.
